packages/hardhat-plugin/test/config.ts

```diff
@@ -1,39 +1,58 @@
 /* eslint-disable import/no-unused-modules */
 
-import { DeployConfig } from "@ignored/ignition-core";
+import { DeployConfig, buildModule } from "@ignored/ignition-core";
 import { assert } from "chai";
 
 import { KeyListOf } from "./type-helper";
 import { useEphemeralIgnitionProject } from "./use-ignition-project";
 
 describe("config", () => {
-  useEphemeralIgnitionProject("with-config");
+  describe("loading", () => {
+    useEphemeralIgnitionProject("with-config");
 
-  let loadedOptions: DeployConfig;
+    let loadedOptions: DeployConfig;
 
-  beforeEach(function () {
-    loadedOptions = this.hre.config.ignition;
-  });
+    beforeEach(function () {
+      loadedOptions = this.hre.config.ignition;
+    });
 
-  it("should apply blockConfirmations", async function () {
-    assert.equal(loadedOptions.blockConfirmations, 10);
-  });
+    it("should apply blockConfirmations", async function () {
+      assert.equal(loadedOptions.blockConfirmations, 10);
+    });
 
-  it("should apply blockPollingInterval", async function () {
-    assert.equal(loadedOptions.blockPollingInterval, 100);
-  });
+    it("should apply blockPollingInterval", async function () {
+      assert.equal(loadedOptions.blockPollingInterval, 100);
+    });
+
+    it("should apply transactionTimeoutInterval", async function () {
+      assert.equal(loadedOptions.transactionTimeoutInterval, 60 * 1000);
+    });
 
-  it("should apply transactionTimeoutInterval", async function () {
-    assert.equal(loadedOptions.transactionTimeoutInterval, 60 * 1000);
+    it("should only have known config", () => {
+      const configOptions: KeyListOf<DeployConfig> = [
+        "blockConfirmations",
+        "blockPollingInterval",
+        "transactionTimeoutInterval",
+      ];
+
+      assert.deepStrictEqual(Object.keys(loadedOptions).sort(), configOptions);
+    });
   });
 
-  it("should only have known config", () => {
-    const configOptions: KeyListOf<DeployConfig> = [
-      "blockConfirmations",
-      "blockPollingInterval",
-      "transactionTimeoutInterval",
-    ];
+  describe("validating", () => {
+    useEphemeralIgnitionProject("with-invalid-config");
+
+    it("should throw when given a `blockConfirmations` value less than 1", async function () {
+      const moduleDefinition = buildModule("FooModule", (m) => {
+        const foo = m.contract("Foo");
+
+        return { foo };
+      });
 
-    assert.deepStrictEqual(Object.keys(loadedOptions).sort(), configOptions);
+      await assert.isRejected(
```

Comment:
I think the deploy function for `useEphemeralIgnitionProject` isn't passing the config from `hre.config.ignition` into the deploy call here.

Reply:
Done.